### PR TITLE
New shared settings and general improvements.

### PR DESCRIPTION
**Pull request by almathew**
## What's this PR do?
A bunch of things. A lot more than it should.
- Adds settings.py to the repository. This replaces the dev_settings.py file most people are using. If you want custom stuff for yourself, they now go in local_settings.py. But there should always be enough in settings.py to run any dev environment. 
- Adds a command, `dkd`. Same as `dkm` but runs microsites by default instead of myjobs.
- Fixes a few issues in tests that either make the tests fail (really, how was that test in seo/tests/test_views.py ever passing?) or prevents them from running with the `--keepdb` flag. So now `--keepdb` should work without fail. 
- Moves some migration-y logic out of the test setup and into the migrations. Because it's migration-y logic.
- A small improvement for tasks, where bad database connections gets closed before the tests are run. We've been seeing issues with bad MySQL connections causing task failures. This should fix some/most of them.

This now means that we should only need to give new devs solr configs and a secrets file on their first day.

## Where should the reviewer start?
settings.py is the biggest of the changes. Are the changes correct, agreeable, and positive?

## How should this be manually tested?
Run tests. They should work for you with no effort (or, well, maybe a bundles/custom.css file because I can't figure that one out).

## Things I learned while doing this
- Sometimes we really do need to be using `TransactionalTestCase` - whenever there's multiple db interactions specifically - but it's really, really slow.
- Solr is really, really slow in dev, which makes running things like the seo tests awful. We might discuss getting away from just blanket inheriting off of base test classes that do things like add to/remove from Solr when the tests aren't even using Solr.

### CODE REVIEW

File: universal/decorators.py [hunk 1]
```diff
@@ -44,25 +65,6 @@ def wrap(request, *args, **kwargs):
         return wraps(view_func)(wrap)
     return decorator
 
-def activate_user(view_func):
```
Comment (almathew) [inline universal/decorators.py]: More unused code.

File: dk2/env.sh [hunk 2]
```diff
@@ -3,11 +3,11 @@ dk() {
 }
 
 dkm() {
-    dk -P settings_myjobs manage "$@"
+    dk manage "$@"
 }
 
 dkr() {
-    dk -P settings_myjobs:. dorun "$@"
+    dk -e myjobs:. dorun "$@"
```
Comment (tdphillips) [inline dk2/env.sh]: `dkd` has `-e ENVIRONMENT=` - does this command need it as well?

File: api/models.py [hunk 3]
```diff
@@ -1,7 +0,0 @@
-from seo.models import User
-from django.db import models
-
-from tastypie.models import create_api_key
-
-models.signals.post_save.connect(create_api_key, sender=User)
```
Comment (almathew) [inline api/models.py]: Moved this to myjobs/models.py in case we still need it for saved search emails.

File: dk2/env.sh (hunk 2 above)
Comment (almathew) [inline dk2/env.sh]: It shouldn't. myjobs is the default.

**Comment by almathew** [comment]
Unfortunately api.py is only nearly unused. There's one stupid admin page that uses it, despite the user being logged in. Not sure it even works for new users, but I guess we'll find out. Not really going to fix it since removing the API endpoints was already outside the scope of what I was doing.

**Pull request merged**